public/wp-content/themes/ccs/functions.php

```diff
@@ -482,4 +482,38 @@ function post_featured_image_and_category_type_json( $data ) {
 	$data->data['category_type'] = get_the_category($data->data['id'])[0]->name;
 
 	return $data;
-  }
\ No newline at end of file
+  }
+
+function getAllHiddenPosts()
+{
+
+	$args = array(
+		'fields'          => 'ids',
+		'numberposts'   => 100,
+		'post_type'		=> 'post',
+		'meta_query'	=> array(
+			'relation'		=> 'AND',
+			array(
+				'key'	  	=> 'Hide_from_View_All',
+				'value'	  	=> '1',
+				'compare' 	=> '=',
+			),
+		),
+	);
+
+	return get_posts($args);
+}
+
+add_action('pre_get_posts', function ($query) {
+
+	$taxArray = array();
+
+	foreach ((array)$query->query["tax_query"] as $each) {
+		array_push($taxArray, $each["taxonomy"]);
+	}
+
+	if (! is_user_logged_in() && $query->query["post_type"] == "post" && $query->query["posts_per_page"] != 100 && !(in_array("products_services", $taxArray) || in_array("sectors", $taxArray))) {
```

Comment:
I feel as though these conditions could be split up over multiple lines to improve readability rather than on a single single. Just a small suggestion.